pkg/virt-controller/watch/pool/pool_test.go

```diff
@@ -827,6 +829,86 @@ var _ = Describe("Pool", func() {
 
 			Expect(testing.FilterActions(&fakeVirtClient.Fake, "patch", "virtualmachines")).To(HaveLen(1))
 		})
+
+		It("should cleanup VMs with autohealing", func() {
+			pool, vm := DefaultPool(1)
+			pool.Spec.Autohealing = pointer.P(true)
+
+			addPool(pool)
+			poolRevision := createPoolRevision(pool)
+			addCR(poolRevision)
+			vm.Name = fmt.Sprintf("%s-0", pool.Name)
+			vm.Status.StartFailure = &v1.VirtualMachineStartFailure{
+				ConsecutiveFailCount: 3,
+			}
+
+			createVMsWithOrdinal(pool, 1, poolRevision, poolRevision, vm)
+
+			sanityExecute()
+
+			vms, err := fakeVirtClient.KubevirtV1().VirtualMachines(pool.Namespace).List(context.TODO(), metav1.ListOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(vms.Items).To(BeEmpty())
+
+			Expect(testing.FilterActions(&fakeVirtClient.Fake, "delete", "virtualmachines")).To(HaveLen(1))
+		})
+
+		It("should cleanup VMs when VM is in not ready state for too long", func() {
+			pool, vm := DefaultPool(1)
+			pool.Spec.Autohealing = pointer.P(true)
+
+			addPool(pool)
+			poolRevision := createPoolRevision(pool)
+			addCR(poolRevision)
+
+			vm.Name = fmt.Sprintf("%s-0", pool.Name)
+			vm.Labels = pool.Spec.VirtualMachineTemplate.ObjectMeta.Labels
+			vm.Labels[virtv1.VirtualMachinePoolRevisionName] = poolRevision.Name
+			vm.Status.PrintableStatus = v1.VirtualMachineStatusErrImagePull
```

Comment:
This could be a DescribeTable over all `PrintableStatuses` you expect the cleanup to occur